### PR TITLE
BUG on CC0071 (introduce field), name clash #266

### PR DESCRIPTION
**Pull request by carloscds**

### CODE REVIEW

File: test/CSharp/CodeCracker.Test/Refactoring/IntroduceFieldFromConstructorTest.cs [hunk 1]
```diff
@@ -128,11 +128,12 @@ namespace ConsoleApplication1
     {
         class TypeName
         {
+            private readonly int par1;
             private readonly int par;
 
             public TypeName(int par)
             {
-               this.par = par;
+               this.par1 = par;
```
Comment (giggio) [inline test/CSharp/CodeCracker.Test/Refactoring/IntroduceFieldFromConstructorTest.cs]: @carloscds This is not supposed to happen. The argument `par` and the field `par` have the same and can be merged together. This test `FieldAlreadyExistsAndMatchesType` verified that.
The bug only happens when you have an event or some other members, like that:

````csharp
class Foo
{
  public event Action a;
  Foo(int a)
  {
  }
}
````

**Comment by carloscds** [comment]
@giggio It's not a proble, look at full code below:

````csharp
class TypeName
        {
            private readonly int par1;
            private readonly int par;
            public TypeName(int par)
            {
               this.par1 = par;
            }
        }
````
Now I will ever create a new var, because I'm checking events, methods, properties and fields. And I'm not checking specifically for types. I removed this test now and update branch.

**Comment by giggio** [comment]
@carloscds This is not how it should go. If the type and the name match, it should simply use the variable, not create a new one.

**Comment by giggio** [comment]
@carloscds Please update to RC1 using current HEAD.

**Comment by carloscds** [comment]
@giggio I will

**Comment by carloscds** [comment]
@giggio Bug Fixed.

**Comment by giggio** [comment]
@carloscds also added some tests and support for inner classes, structs, interfaces, enums, and delegates.

**Pull request merged**